api/base/filters.py

```diff
@@ -49,31 +85,38 @@ def get_query_from_request(self):
 
     def query_params_to_odm_query(self, query_params):
         """Convert query params to a modularodm Query object."""
+
         fields_dict = query_params_to_fields(query_params)
         if fields_dict:
             query_parts = [
-                Q(self.convert_key(key, value), 'eq', self.convert_value(key, value)) for key, value in fields_dict.items()
+                Q(self.convert_key(key=key), self.get_comparison_operator(key=key), self.convert_value(value=value))
+                for key, value in fields_dict.items() if self.is_filterable_field(key=key)
             ]
-            query = functools.reduce(intersect, query_parts)
+            # TODO Ensure that if you try to filter on an invalid field, it returns a useful error.
+            try:
+                query = functools.reduce(intersect, query_parts)
+            except TypeError:
+                query = None
         else:
             query = None
         return query
 
     # Used so that that queries by _id will work
-    def convert_key(self, key, value):
-        if key.strip() == 'id':
-            return '_id'
+    def convert_key(self, key):
+        key = key.strip()
+        if self.serializer_class._declared_fields[key].source:
+            return self.serializer_class._declared_fields[key].source
         return key
 
     # Used to convert string values from query params to Python booleans when necessary
-    def convert_value(self, key, val):
-        val = val.strip()
-        if val in self.TRUTHY:
+    def convert_value(self, value):
```

Comment:
I think included both `key` in the signature for `convert_value` because I had thought of a use case for transforming the value based on the `key`. I can't think of that use case off the top of my head ATM, so I think this is OK for now.